pl/lessons/basics/documentation.md

```diff
@@ -0,0 +1,347 @@
+
+Examples
+
+    iex> Greeter.hello("Sean")
+    "Hello, Sean"
+
+    iex> Greeter.hello("pete")
+    "Hello, pete"
+
+iex>
+```
+
+Zauważ jak za pomocą znaczników, umożliwiliśmy terminalowi sformatowanie dokumentacji. Poza tym, że fajnie to
+wygląda i jest nowatorskim rozwiązaniem w ekosystemie Elixira, to daje nam duże możliwości przy generowaniu HTML-a z
+pomocą narzędzia ExDoc.
+
+## ExDoc
+
+ExDoc jest oficjalnym projektem zespołu Elixira służącym do **generowania HTML-a (HyperText Markup Language) i 
+dokumentacji online dla projektów tworzonych w Elixirze**, które można znaleźć na [GitHub](https://github
+.com/elixir-lang/ex_doc). Na początek stwórzmy nowy projekt w mix-ie:
+
+```bash
+$ mix new greet_everyone
+
+* creating README.md
+* creating .gitignore
+* creating mix.exs
+* creating config
+* creating config/config.exs
+* creating lib
+* creating lib/greet_everyone.ex
+* creating test
+* creating test/test_helper.exs
+* creating test/greet_everyone_test.exs
+
+Your Mix project was created successfully.
+You can use "mix" to compile it, test it, and more:
+
+    cd greet_everyone
+    mix test
+
+Run "mix help" for more commands.
+
+$ cd greet_everyone
+
+```
+
+Następnie skopiujmy adnotację `@doc` do pliku `lib/greeter.ex` i upewnijmy się, że nadal działa on z linii poleceń. 
+Jako że pracujemy z mixem uruchommy IEx korzystając z polecenia `iex -S mix` i wpiszmy:
+
+```bash
+iex> h Greeter.hello
+
+                def hello(name)
+
+Prints a hello message
+
+Parameters
+
+  • name: String that represents the name of the person.
+
+Examples
+
+    iex> Greeter.hello("Sean")
+    "Hello, Sean"
+
+    iex> Greeter.hello("pete")
+    "Hello, pete"
+```
+
+### Instalacja
+
+Assuming all is well, and we're seeing the output above suggests that we are ready to set up ExDoc. Within our `mix.exs` file add the two required dependencies to get started; `:earmark` and `:ex_doc`.
+Wnioskując z powyższego komunikatu jesteśmy już gotowi by skonfigurować ExDoc. W pliku `mix.exs` musimy dodać dwie 
+zależności `:earmark` i `:ex_doc`.
+
+```elixir
+  def deps do
+    [{:earmark, "~> 0.1", only: :dev},
+    {:ex_doc, "~> 0.11", only: :dev}]
+  end
+```
+
+Podając opcję `only: :dev` mówimy mixowi, że nie chcemy pobierać i kompilować tych zależności na środowisku innym niż
+ deweloperskie. Ale czym jest Earmark? Earmark jest to parser znaczników napisany dla Elixira, który służy ExDocowi 
+ na zamianę dokumentacji z adnotacji `@moduledoc` i `@doc` na elegancki kod HTML.
+
+Oczywiście nie jesteśmy ograniczeni tylko do Earmarka. Jak chcesz możesz zamienić go na Pandoc, Hoedown albo Cmark; 
+to wymaga trochę dodatkowej konfiguracji, o której poczytasz [tutaj (ang.)](https://github
+.com/elixir-lang/ex_doc#changing-the-markdown-tool). Na potrzeby tej lekcji pozostaniemy jednak przy Earmarku.
+
+### Generowanie dokumentacji
+
+Idąc dalej w linii poleceń wydajemy dwie komendy:
+
+```bash
+$ mix deps.get # gets ExDoc + Earmark.
```

Comment:
# gets ExDoc + Earmark. -> # pobiera ExDoc + Earmark.
# makes the documentation. -> # tworzy dokumentację.

I think those comments can be translatated as they are just informative for this lesson.